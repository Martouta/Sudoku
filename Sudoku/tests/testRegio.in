9

1

2

3

5
-1

5
9

5
8

7
-1

7
9

7
3

4
3
2

6
6
2

6
7
2

3

2

1

7
6

3

2

0



Explicación:


Escribe el tamano de la region

9 //indicamos el tamaño de la región (número de casillas que forman parte de esta región)

1 //obtenemos el número de celdas totales para ver que coinciden con el número introducido previamente

2 //en este caso obtenemos el número de celdas rellenas, que en esta ocasión es 0 ya que aún no hemos introducido ningún valor

3 //para confirmar el caso anterior, printamos qué celdas dentro de la región están llenas o vacías

5 //intentamos coger el valor de una posición no válida (la -1), y nos salta una excepción indicando que hemos accedido a una posición fuera de rango con el código -2
-1

5 //igual que en el caso anterior, no se cumple que la posición pos introducida 0<=pos<numTotalCeldasRegion, y nos salta la misma excepción
9

5 //en este caso si que la posición es válida, pero nos devuelve un -1 ya que la celda no contiene ningún valor aún
8

7 //intentamos borrar una posición no válida, igual que antes
-1


7 //intentamos borrar una posición no válida, igual que antes
9

7 //eliminamos el valor de una celda que no contenía ningún valor, para mostrar que no pasa nada
3

4 //obtenemos la celda número 3 (dentro de la región), y hacemos pruebas con ella añadiendole un valor introduciod (2 en este caso) y borrandolo, mostrándolo por pantalla
3
2

6 //introducimos el valor 2 a la celda 6 para mostrar que se introduce correctamente
6
2

6 //igual que el caso anterior
7
2

3 //mostramos que celdas están vacías y cuales no lo están, así comprobando que se han actualizado las celdas rellenas

2 //igual que el caso anterior, nos sirve para comprobar que hemos actualizado las celdas rellenas

1 //nos indica las celdas totales para ver que no lo hemos modificado erróneamente

7 //probamos de eliminar el valor de celdas con un valor previamente añadido, y así comprobar que lo elimina bien
6

3 //mostramos que celdas están vacías para comprobar que se ha ejecutado correctamente el caso anterior

2 //igual que antes, miramos que se ha actualizado correctamente el número de celdas rellenas

0