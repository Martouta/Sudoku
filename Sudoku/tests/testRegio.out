9

0

true
true
true
true
true
true
true
true
true

La posicion esta fuera del rango permitido
-2

La posicion esta fuera del rango permitido
-2

-1

La posicion esta fuera del rango permitido
La posicion esta fuera del rango permitido
-2

La posicion esta fuera del rango permitido
La posicion esta fuera del rango permitido
-2

-1

2
-1

2

2

true
true
true
true
true
true
false
false
true

2

9

-1

true
true
true
true
true
true
true
false
true

1